### PR TITLE
Enable strict option in load_npz

### LINKED ISSUE #2599: Add strict option in load_npz
https://github.com/pfnet/chainer/blob/master/chainer/serializers/npz.py#L118
`serializers.load_npz` can not use useful `strict` option of `serializers.NpzDeserializer` just due to the absence in its input arguments. We have to add it.

### PR DESCRIPTION
**Pull request by mitmul**
Fix #2599

### CODE REVIEW

**Comment by unnonouno** [comment]
Write a test.

**Comment by mitmul** [comment]
I'm very sorry about that. I added test cases for `strict=False`.

File: tests/chainer_tests/serializers_tests/test_npz.py [hunk 1]
```diff
@@ -244,14 +244,22 @@ def tearDown(self):
...
         self.assertEqual(obj.serialize.call_count, 1)
         (serializer,), _ = obj.serialize.call_args
         self.assertIsInstance(serializer, npz.NpzDeserializer)
 
+    def test_load_without_strict(self):
+        obj = mock.MagicMock()
+        npz.load_npz(self.temp_file_path, obj, False)
+
+        self.assertEqual(obj.serialize.call_count, 1)
+        (serializer,), _ = obj.serialize.call_args
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_npz.py]: Check argument.

File: tests/chainer_tests/serializers_tests/test_npz.py [hunk 2]
```diff
@@ -332,5 +340,15 @@ def test_load_optimizer(self):
         for param in self.parent.params():
             self.assertTrue((param.data == 1).all())
 
+    def test_load_optimizer_without_strict(self):
+        for param in self.parent.params():
+            param.data.fill(1)
+        npz.save_npz(self.temp_file_path, self.parent, self.compress)
+        for param in self.parent.params():
+            param.data.fill(0)
+        npz.load_npz(self.temp_file_path, self.parent, False)
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_npz.py]: Test behavior which changes only when `strict=False`

**Comment by mitmul** [comment]
@unnonouno Thank you for your review, I fixed the tests. Could you check it again?

File: tests/chainer_tests/serializers_tests/test_npz.py [hunk 3]
```diff
@@ -269,17 +269,27 @@ def tearDown(self):
...
 
+    def test_load_without_strict(self):
+        obj = mock.MagicMock()
+        npz.load_npz(self.temp_file_path, obj, strict=False)
+
+        self.assertEqual(obj.serialize.call_count, 1)
+        (serializer,), _ = obj.serialize.call_args
+        self.assertFalse(serializer.strict)
+        self.assertIsInstance(serializer, npz.NpzDeserializer)
+        self.assertEqual(serializer.strict, False)
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_npz.py]: This is equivalent to line 287.

File: tests/chainer_tests/serializers_tests/test_npz.py [hunk 4]
```diff
@@ -269,17 +269,27 @@ def tearDown(self):
...
             os.remove(self.temp_file_path)
 
-    def test_load(self):
+    def test_load_with_strict(self):
         obj = mock.MagicMock()
         npz.load_npz(self.temp_file_path, obj)
 
         self.assertEqual(obj.serialize.call_count, 1)
         (serializer,), _ = obj.serialize.call_args
         self.assertIsInstance(serializer, npz.NpzDeserializer)
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_npz.py]: Check if serializer.strict is True.

File: tests/chainer_tests/serializers_tests/test_npz.py (hunk 3 above)
Comment (mitmul) [inline tests/chainer_tests/serializers_tests/test_npz.py]: Thank you for pointing out it. I removed this line.

File: tests/chainer_tests/serializers_tests/test_npz.py (hunk 4 above)
Comment (mitmul) [inline tests/chainer_tests/serializers_tests/test_npz.py]: Thank you for pointing out it. I added one line to check `serializer.strict` is True.

**Comment by unnonouno** [comment]
LGTM. Please rebase or merge the master.

**Comment by mitmul** [comment]
@unnonouno Sorry for my delay. Merged with master

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**